inference-engine/src/gna_plugin/gna_plugin.cpp

```diff
@@ -1132,7 +1132,11 @@ InferenceEngine::IExecutableNetwork::Ptr GNAPlugin::ImportNetwork(const std::str
         THROW_GNA_EXCEPTION << "Cannot open file to import model: " << modelFileName;
     }
 
-    auto header = GNAModelSerial::ReadHeader(inputStream);
+    return ImportNetwork(inputStream);
+}
+
+InferenceEngine::IExecutableNetwork::Ptr GNAPlugin::ImportNetwork(std::istream& networkModel) {
```

Comment:
overload at GNAPlugin level might not need - you can move this stream init into gna_executable network, this makes inferenceEngine->GNAPlugin API simpler.